include/hip/devicelib/double_precision/dp_math.hh

```diff
@@ -47,6 +78,41 @@ static inline __device__ long int lround(double x) {
 static inline __device__ long long int llrint(double x) { return lrint(x); }
 static inline __device__ long long int llround(double x) { return lround(x); }
 
+static inline __device__ double nearbyint(double x) {
+  return __builtin_nearbyint(x);
+}
+
+namespace std {
+// Clang does provide device side std:: functions via HIP include
+// wrappers but, alas, the wrappers won't compile on CHIP-SPV due to
+// presence of AMD specific built-ins.
+using ::acos;
+using ::asin;
+using ::atan;
+using ::ceil;
+using ::copysign;
+using ::cos;
+using ::cosh;
+using ::erf;
+using ::erfc;
+using ::exp;
+using ::expm1;
+using ::floor;
+using ::lgamma;
+using ::log;
+using ::log10;
+using ::log1p;
+using ::log2;
+using ::nearbyint;
+using ::nextafter;
+using ::sin;
+using ::sinh;
+using ::sqrt;
+using ::tan;
+using ::tanh;
+using ::trunc;
+} // namespace std
+
```

Comment:
I'm confused here. 

> Clang does provide device side std:: functions via HIP include wrappers but, alas, the wrappers won't compile

don't these `using` statements essentially map `cos` to `std::cos` which you just said don't compile?

Reply:
It's other way around. Call to std::XYZ will resolve to XYZ in root namespace (aka. to a function which is not defined within namespace clause).